net/pubsub.go

```diff
@@ -0,0 +1,165 @@
+package net
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/textileio/go-threads/core/thread"
+	pb "github.com/textileio/go-threads/net/pb"
+)
+
+// TopicHandler receives all pushed thread records.
+type TopicHandler func(*pb.PushRecordRequest)
+
+// TopicManager manages thread pubsub topics.
+type TopicManager struct {
```

Comment:
Manager hold all active thread topics, their peer event handlers, and subscriptions.